src/operators.jl

```diff
@@ -235,8 +236,8 @@ Flatten a signal of signals into a signal which holds the
 value of the current signal. The `typ` keyword argument specifies
 the type of the flattened signal. It is `Any` by default.
 """
-function flatten(input::Signal; typ=Any)
-    n = Signal(typ, value(value(input)), (input,))
+function flatten(input::Signal; typ=Any, name=auto_name())
+    n = Signal(typ, value(value(input)), (input,); name=name)
```

Comment:
I wonder if instead of adding keyword arguments to everything we can define `name!(signal, "name")` to set the name for signals you want to name.

It might also be nicer to default have default names such as "map(signal-1, signal-2)-1", "filter(signal-1)-2" etc.